apps/arweave/c_src/vdf/ar_vdf_nif.c

```diff
@@ -137,7 +146,47 @@ static ERL_NIF_TERM vdf_sha2_exp_nif(ErlNifEnv* envPtr, int argc, const ERL_NIF_
 	size_t outCheckpointSize = VDF_SHA_HASH_SIZE*checkpointCount;
 	ERL_NIF_TERM outputTermCheckpoint;
 	unsigned char* outCheckpoint = enif_make_new_binary(envPtr, outCheckpointSize, &outputTermCheckpoint);
-	vdf_sha2_exp_ptr(Salt.data, Seed.data, temp_result, outCheckpoint, checkpointCount, skipCheckpointCount, hashingIterations);
+	vdf_sha2_fused_ptr(Salt.data, Seed.data, temp_result, outCheckpoint, checkpointCount, skipCheckpointCount, hashingIterations);
```

Comment:
What do you think about refactoring the body of `vdf_sha2_fused_ptr` and `vdf_sha2_fused_nif` to eliminate the duplicate code? It looks like its onlyt he `xxx_ptr` that differs between the two? So maybe that could be an argument to the refactored helper function.